src/tools/concurrency/TracingBackend.java

```diff
@@ -79,21 +80,16 @@
   private static final ArrayBlockingQueue<byte[]> emptyBuffers =
       new ArrayBlockingQueue<>(BUFFER_POOL_SIZE);
 
-  private static final ArrayBlockingQueue<BufferAndLimit> fullBuffers =
-      new ArrayBlockingQueue<>(BUFFER_POOL_SIZE);
-
-  private static final java.util.concurrent.ConcurrentLinkedQueue<Object[]> externalData =
-      new java.util.concurrent.ConcurrentLinkedQueue<Object[]>();
-
   // contains symbols that need to be written to file/sent to debugger,
   // e.g. actor type, message type
   private static final ArrayList<SSymbol> symbolsToWrite =
       new ArrayList<>();
 
   private static FrontendConnector front = null;
 
-  private static long              collectedMemory = 0;
+  private static long collectedMemory = 0;
   private static TraceWorkerThread workerThread;
```

Comment:
Please distinguish the purposes of these two worker threads. It's unclear why we need two. Numbering doesn't help.

Reply:
I renamed them to make the purposes more clear and added a description comment. 

  /**
   * Used for writing the traces in normal actor Tracing.
   * When snapshoting, this points to the worker responsible for the trace
   * associated with the current snapshot.
   */
  private static TraceWorkerThread currentWorkerThread;
  /**
   * When snapshoting, this points to the worker that handles remaining
   * trace data belonging to the previous snapshot.
   */
  private static TraceWorkerThread previousWorkerThread;